crates/uv/src/commands/build.rs

```diff
@@ -0,0 +1,338 @@
+        &client_builder,
+        cache,
+        Some(&PythonDownloadReporter::single(printer)),
+    )
+    .await?
+    .into_interpreter();
+
+    // Add all authenticated sources to the cache.
+    for url in index_locations.urls() {
+        store_credentials_from_url(url);
+    }
+
+    // Initialize the registry client.
+    let client = RegistryClientBuilder::new(cache.clone())
+        .native_tls(native_tls)
+        .connectivity(connectivity)
+        .index_urls(index_locations.index_urls())
+        .index_strategy(index_strategy)
+        .keyring(keyring_provider)
+        .allow_insecure_host(allow_insecure_host.to_vec())
+        .markers(interpreter.markers())
+        .platform(interpreter.platform())
+        .build();
+
+    // Determine whether to enable build isolation.
+    let environment;
+    let build_isolation = if no_build_isolation {
+        environment = PythonEnvironment::from_interpreter(interpreter.clone());
+        BuildIsolation::Shared(&environment)
+    } else if no_build_isolation_package.is_empty() {
+        BuildIsolation::Isolated
+    } else {
+        environment = PythonEnvironment::from_interpreter(interpreter.clone());
+        BuildIsolation::SharedPackage(&environment, no_build_isolation_package)
+    };
+
+    // TODO(charlie): These are all default values. We should consider whether we want to make them
+    // optional on the downstream APIs.
+    let build_constraints = [];
+    let hasher = HashStrategy::None;
+
+    // Resolve the flat indexes from `--find-links`.
+    let flat_index = {
+        let client = FlatIndexClient::new(&client, cache);
+        let entries = client.fetch(index_locations.flat_index()).await?;
+        FlatIndex::from_entries(entries, None, &hasher, build_options)
+    };
+
+    // Initialize any shared state.
+    let state = SharedState::default();
+
+    // Create a build dispatch.
+    let build_dispatch = BuildDispatch::new(
+        &client,
+        cache,
+        &build_constraints,
+        &interpreter,
+        index_locations,
+        &flat_index,
+        &state.index,
+        &state.git,
+        &state.in_flight,
+        index_strategy,
+        config_setting,
+        build_isolation,
+        link_mode,
+        build_options,
+        exclude_newer,
+        sources,
+        concurrency,
+    );
+
+    // Determine the build plan from the command-line arguments.
+    let path = path.unwrap_or_else(|| &*CWD);
+    let output_dir = path.join("dist");
+    fs_err::tokio::create_dir_all(&output_dir).await?;
+
+    let plan = match (sdist, wheel) {
+        (false, false) => BuildPlan::SdistToWheel,
+        (true, false) => BuildPlan::Sdist,
+        (false, true) => BuildPlan::Wheel,
+        (true, true) => BuildPlan::SdistAndWheel,
+    };
+
+    // Prepare some common arguments for the build.
+    let subdirectory = None;
+    let version_id = path.user_display().to_string();
+    let dist = None;
+
+    let assets = match plan {
```

Comment:
Do we want to show "Building wheel"? I think so -- should it include a progress spinner? (Also think so.)

Do we want to show the build _output_ by default? (As of #6903 it will show in `--verbose`.)